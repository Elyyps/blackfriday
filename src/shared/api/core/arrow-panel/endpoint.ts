import { generateArrowPanelData } from "./generate-dummy-data";

export const generateTabsDummyData = generateArrowPanelData();
