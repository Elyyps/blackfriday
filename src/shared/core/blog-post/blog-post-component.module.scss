.blog-post {
  display: flex;
  align-items: center;
  font-size: 14px;
  &__title {
    font-family: $global-font-family-secondary;
    color: $gray-dark;
    font-size: 16px;
    font-weight: bold;
    margin-bottom: 8px;
    @media (max-width: $breakpoint-small-max) {
      font-size: 14px;
      margin-bottom: 0;
    }
  }
  &__title:hover {
    color: $global-link-color;
  }
  &__body {
    margin: auto;
    color: $gray-light;
  }
  &__image {
    width: 150px;
    height: 105px;
    margin-right: 24px;
    flex: 0 0 auto;
    overflow: hidden;
    position: relative;
    border-radius: 2px;
    @media (max-width: $breakpoint-small-max) {
      margin-right: 16px;
    }
    img,
    svg {
      width: 100%;
      height: 100%;
    }
  }
  &__link {
    color: $global-link-color;
    font-weight: 600;
  }
}
