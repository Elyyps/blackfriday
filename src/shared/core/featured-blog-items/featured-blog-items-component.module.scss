.featured-blog-items {
  &__card {
    display: block;
    position: relative;
    color: $global-inverse-color;
    text-decoration: none;
  }
  &__card:hover {
    img {
      transform: scale(1.05);
    }
    .featured-blog-items__card-image:after {
      background-image: linear-gradient(
        to bottom,
        rgba(28, 28, 28, 0) 0,
        rgba(28, 28, 28, 0.7) 77%,
        rgba(28, 28, 28, 0.8)
      );
    }
  }
  &__card-image {
    height: 240px;
    width: 100%;
    border-radius: $global-border-radius;
    overflow: hidden;
    position: relative;
    @media (max-width: $breakpoint-medium-max) {
      height: 180px;
    }

    &:after {
      position: absolute;
      left: 0;
      right: 0;
      top: 0;
      bottom: 0;
      content: "";
      background-image: linear-gradient(
        to bottom,
        rgba(28, 28, 28, 0) 0,
        rgba(28, 28, 28, 0.6) 77%,
        rgba(28, 28, 28, 0.7)
      );
    }

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
      transition: transform 0.5s ease;
    }
  }

  &__card-title {
    position: absolute;
    left: 32px;
    right: 32px;
    bottom: 32px;
    color: $global-inverse-color;
    margin: 0;
    @media (max-width: $breakpoint-medium-max) {
      left: 16px;
      right: 16px;
      bottom: 16px;
      font-size: 16px;
    }
  }
}
