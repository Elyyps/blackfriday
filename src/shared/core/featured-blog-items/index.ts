export * from "./featured-blog-items.component";
