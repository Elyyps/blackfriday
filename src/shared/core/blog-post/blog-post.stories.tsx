import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { BlogPostComponent } from "./blog-post.component";
import { generateBlogPostData } from "@app/api/core/blog-post/endpoint";
const posts = generateBlogPostData();

storiesOf("BlogPost", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <BlogPostComponent posts={posts[1]} />);
