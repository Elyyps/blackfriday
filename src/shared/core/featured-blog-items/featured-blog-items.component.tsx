import * as React from "react";
import styles from "./featured-blog-items-component.module.scss";
import { LinkComponent } from "../link";
import { ImageComponent } from "../image";
import { BlogPost } from "@app/api/core/blog-post/blog-post";

export interface IFeaturedBlogItemsComponentProps {
  item: BlogPost;
}

const FeaturedBlogItemsComponent = (props: IFeaturedBlogItemsComponentProps) => (
  <div className={styles["featured-blog-items"]}>
    <div className={styles["featured-blog-items__card"]}>
      <LinkComponent to={props.item.link.url}>
        <div className={styles["featured-blog-items__card-image"]}>
          <ImageComponent alt={props.item.title} src={props.item.image} />
        </div>
        <h3 className={styles["featured-blog-items__card-title"]}>{props.item.title}</h3>
      </LinkComponent>
    </div>
  </div>
);

export { FeaturedBlogItemsComponent };
