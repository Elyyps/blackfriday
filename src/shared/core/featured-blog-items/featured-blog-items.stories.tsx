import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { FeaturedBlogItemsComponent } from "./featured-blog-items.component";
import { generate2BlogPostsDataBelgian } from "@app/api/core/blog-post/endpoint";
const item = generate2BlogPostsDataBelgian();
storiesOf("FeaturedBlogItems", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <FeaturedBlogItemsComponent item={item[0]} />);
