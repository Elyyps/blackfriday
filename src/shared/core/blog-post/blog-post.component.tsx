import * as React from "react";
import styles from "./blog-post-component.module.scss";
import { LinkComponent } from "../link";
import { ImageComponent } from "@app/prep/modules-prep/core";
import { BlogPost } from "@app/api/core/blog-post/blog-post";

export interface IBlogPostComponentProps {
  posts: BlogPost;
}

const BlogPostComponent = (props: IBlogPostComponentProps) => (
  <div className={styles["blog-post"]}>
    <LinkComponent to={props.posts.link}>
      {props.posts.image && (
        <div className={styles["blog-post__image"]}>
          <ImageComponent src={props.posts.image} />
        </div>
      )}
      <div className={styles["blog-post__body"]}>
        <div className={styles["blog-post__title"]}>{props.posts.title}</div>
        <div className="uk-visible@m">
          {props.posts.content}
          <span className={styles["blog-post__link"]}>{props.posts.linkTitle}</span>
        </div>
      </div>
    </LinkComponent>
  </div>
);

export { BlogPostComponent };
