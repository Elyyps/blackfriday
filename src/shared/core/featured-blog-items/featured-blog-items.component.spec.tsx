import React from "react";
import { shallow } from "enzyme";
import { FeaturedBlogItemsComponent } from "./featured-blog-items.component";
import { generate2BlogPostsDataBelgian } from "@app/api/core/blog-post/endpoint";
const item = generate2BlogPostsDataBelgian();
describe("[FeaturedBlogItems]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<FeaturedBlogItemsComponent item={item[0]} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
