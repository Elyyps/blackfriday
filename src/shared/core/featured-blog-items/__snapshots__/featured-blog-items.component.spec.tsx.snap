// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FeaturedBlogItems] should render component without crashing 1`] = `
<div
  className="featured-blog-items"
>
  <div
    className="featured-blog-items__card"
  >
    <LinkComponent
      to="/blog-single"
    >
      <div
        className="featured-blog-items__card-image"
      >
        <ImageComponent
          src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/blog-single.png"
        />
      </div>
      <h3
        className="featured-blog-items__card-title"
      >
        De beste Moederdag cadeau deals
      </h3>
    </LinkComponent>
  </div>
</div>
`;
